### PR TITLE
fix Pose2 preambleCache bug

### PR DESCRIPTION
**Pull request by dehann**
The idea here is to try reduce hot loop memory consumption.  

So just quickly trying on hexagonal example...

Once upon a time, hex would take at best around 30s to solve...  After this PR, and on a computer thermally throttling to around 2.8GHz, hex solves in less than 8 seconds.  Usual caveats of all processes having JIT compiled etc. (that's using 5 processes using Distributed.jl).  Haven't had time to test properly (i've been lazy with fg in global scope).  

TLDR; i expect this PR to reduce the hot loop memory consumption.  Proper testing will tell how much this PR helps, but it seems to be helping somewhere between a little towards a lot.

### CODE REVIEW

**Comment by dehann** [comment]
ah, should put q_hat in the cache as well...  will try do tomorrow quick

**Comment by jim-hill-r** [comment]
Excited for the performance gains! This looks like a side-by-side caching opposed to in-line caching. Just curious about the design choice there.

**Review by jim-hill-r — APPROVED** [review]

**Comment by dehann** [comment]
PS, I added this to the CJL collection of high level design decisions: https://github.com/JuliaRobotics/Caesar.jl/wiki/High-Level-Requirements

I'm now starting to think we should consolidate that with the RequirementsManagement system.
